text_generation/causal_lm/cpp/continuous_batching/library/src/sequence_group.hpp

```diff
@@ -448,7 +449,15 @@ class SequenceGroup {
             // TO DO: Now we always stream for greedy search for the sake of benchmarking
             if (true /* m_sampling_params.stream */) {
                 for (auto& sequence : m_sequences) {
-                        outputs.emplace(sequence->get_grouped_id(), sequence->get_last_generation_output(m_sampling_params));
+                    if (!sequence->has_finished()) {
+                        const auto last_gen_token = sequence->get_last_generation_output();
+                        outputs.emplace(sequence->get_grouped_id(), last_gen_token);
+                        if (m_sampling_params.max_new_tokens == sequence->get_generated_len() ||
+                            last_gen_token.generated_token_ids.back() == m_sampling_params.eos_token_id && !m_sampling_params.ignore_eos) {
+                            // stop sequence by max_new_tokens or EOS token
+                            sequence->set_status(SequenceStatus::FINISHED);
```

Comment:
let's move changing sequence status to sampler code

for cases with multiple sequences within a group, we can consider streaming = false

Reply:
Done